src/Classes/Jobs/PurgeCommentsJob.php

```diff
@@ -0,0 +1,102 @@
+<?php
+
+namespace CurseProfile\Classes\Jobs;
+
+use CurseProfile\Classes\Comment;
+use CurseProfile\Classes\CommentBoard;
+use Fandom\Includes\Logging\Loggable;
+use Fandom\Includes\Rabbit\JobConfigurator;
+use Fandom\Includes\Rabbit\JobParams;
+use Job;
+use JobSpecification;
+use MediaWiki\Config\ServiceOptions;
+use MediaWiki\MediaWikiServices;
+use MediaWiki\Title\Title;
+use MediaWiki\User\UserFactory;
+use Wikimedia\Rdbms\ILoadBalancer;
+use Wikimedia\Rdbms\IResultWrapper;
+
+/**
+ * Job that mass purges comments made by a user
+ */
+class PurgeCommentsJob extends Job {
+	use Loggable;
+
+	private const CONSTRUCTOR_OPTIONS = [ 'FandomBotUserName' ];
+
+	private const COMMAND = "CurseProfile\\PurgeComments";
+	private const JOB_OWNER = JobConfigurator::UGC_TEAM_LABEL;
+	private int $targetUserId;
+	private string $summary;
+	private string $time;
+
+	public function __construct(
+		array $params,
+		private ILoadBalancer $lb,
+		private UserFactory $userFactory,
+		private ServiceOptions $serviceOptions
+	) {
+		$this->serviceOptions->assertRequiredOptions( self::CONSTRUCTOR_OPTIONS );
+		$this->targetUserId = $params['targetUserId'];
+		$this->summary = $params['summary'];
+		$this->time = $params['time'];
+
+		parent::__construct( self::COMMAND, $params );
+	}
+
+	public static function newSpecification( int $targetUserId, string $summary, string $time ): JobSpecification {
+		return new JobSpecification(
+			self::COMMAND,
+			[
+				JobParams::JOB_OWNER => self::JOB_OWNER,
+				'targetUserId' => $targetUserId,
+				'summary' => $summary,
+				'time' => $time,
+			]
+		);
+	}
+
+	public static function newInstance( ?Title $title, $params ): Job {
+		$services = MediaWikiServices::getInstance();
+
+		return new PurgeCommentsJob(
+			$params,
+			$services->getDBLoadBalancer(),
+			$services->getUserFactory(),
+			new ServiceOptions(
+				self::CONSTRUCTOR_OPTIONS,
+				$services->getMainConfig()
+			),
+		);
+	}
+
+	public function run(): void {
+		$botUser = $this->userFactory->newFromName( $this->serviceOptions->get( 'FandomBotUserName' ) );
+
+		$commentsFromDB = $this->getCommentsFromTimestamp( $this->targetUserId, $this->time );
+		foreach ( $commentsFromDB as $commentRow ) {
+			try {
+				$comment = new Comment( json_decode( json_encode( $commentRow ), true ) );
+				CommentBoard::purgeComment( $comment, $botUser, $this->summary );
+			} catch ( \MWException $e ) {
+				$this->error( "Failed to purge comment", [ 'exception' => $e ] );
+			}
```

Comment:
I think it would be good to batch the query and wait for replication between each batch, as in https://github.com/Wikia/unified-platform/pull/15590

In a Job context, you will need `getEmptyTransactionTicket` + `commitAndWaitForReplication` to make it work.

Reply:
Set the limit to 100